pkg/operator/controller/ingress/status.go

```diff
@@ -728,10 +729,15 @@ func checkDefaultCertificate(secret *corev1.Secret, domain string) error {
 		if cert.Subject.CommonName == domain && !foundSAN {
 			return fmt.Errorf("certificate in secret %s/%s has legacy Common Name (CN) but has no Subject Alternative Name (SAN) for domain: %s", secret.Namespace, secret.Name, domain)
 		}
-		switch cert.SignatureAlgorithm {
-		case x509.SHA1WithRSA, x509.ECDSAWithSHA1:
-			return fmt.Errorf("certificate in secret %s/%s has weak SHA1 signature algorithm: %s (see https://docs.openshift.com/container-platform/4.16/release_notes/ocp-4-16-release-notes.html#ocp-4-16-sha-haproxy-support-removed_release-notes for more details)", secret.Namespace, secret.Name, cert.SignatureAlgorithm)
+		// Prevent the upgrade only if the leaf certificate (the first certificate) uses a SHA1 signature algorithm.
+		// SHA1 can still be used by other certificates in the chain, such as the root and intermediate certificates.
+		if certIndex == 0 {
```

Comment:
I tried to test this, but I couldn't work around this error while creating a TLS secret with improperly ordered certificate:
```
# reverse order
cat exampleca.crt example.crt > example-combo-reversed.crt
$ oc create secret -n openshift-ingress tls router-cert --key=example.key --cert=example-combo-reversed.crt --validate=false
error: tls: private key does not match public key
```

However, in case there is a way to force a CA as index 0 that I am unaware, generally it seems more resilient to do `if !cert.IsCA` to handle any possible edge case.

Thanks! Updated.